src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -9195,7 +9435,7 @@ instruct sarL_rReg_imm(rRegL dst, immI8 shift, rFlagsReg cr)
 %}
 
 // Arithmetic Shift Right by 8-bit immediate
-instruct sarL_mem_imm(memory dst, immI8 shift, rFlagsReg cr)
+instruct sarL_mem_imm(memory dst, immI shift, rFlagsReg cr)
```

Comment:
Why this change to type of constant?

Reply:
For other shift nodes, the `Ideal` method clips the constant shift count to be in the correct range so we can match `immI8` with it. The `RShiftLNode` does not have an `Ideal` method so we have to do that in the backend. Previously, constant shifts that out-of-bounds for 8-bit immediate still match with the variable shift rule, but as I exclude constant shifts from `sarL_rReg_rReg`, this reveals.
Thank you very much.

Reply:
okay.